lib/routes/v1.js

```diff
@@ -169,6 +183,18 @@ api.declare({
 
   let modDate = new Date();
 
+  // We want to make sure that all AMIs that we are submitting are valid
+  for (let region of input.regions) {
+    let validAmi = await amiExists(this.ec2, region.launchSpec.ImageId, region.region);
```

Comment:
I would do:
```js
let missing = [];
await Promise.all(input.regions.map(async (def) => {
  if (await amiExists(this.ec2, def.launchSpec.ImageId, def.region)) {
    missing.push({imageId: def.launchSpec.ImageId, region: def.region});
  }
}));
if (missing.length > 0) {
   .... // Send error
}
```
or even shorter:
```js
let missingAMI = _.every(await Promise.all(input.regions.map(def => {
  return amiExists(this.ec2, def.launchSpec.ImageId, def.region);
})));
if (missingAMI) {
   .... // Send error without saying which AMI is missing...
}
```

Either way, using `Promise.all` makes sense as we get the requests in parallel.. As the requests goes out to different regions, there is no reason not to make them in parallel.